### ISSUE TITLE
[Inductor-CPU] qlinear_binary output may have strides whose value can't be determined

### ISSUE THREAD

**Issue by sanchitintel** [pytorch/pytorch#148199]
### 🐛 Describe the bug

When `M` dimension is 1 and dynamic shape support is enabled with `torch.compile`, `qlinear_binary` op's output's outermost dim's stride may have a symbolic value that can't be determined.

Please copy-paste this UT in `test/inductor/test_mkldnn_pattern_matcher.py`:

```python
    @skipIfNoDynamoSupport
    @skipIfNoONEDNN
    # TODO: investigate options of torch.compile in fbcode
    @unittest.skipIf(IS_FBCODE, "Failing in fbcode")
    @parametrize("has_bias", [False,])
    @parametrize("dtype", [torch.bfloat16,])
    @parametrize("dynamic", [True,])
    @parametrize("reshape_a", [True,])
    @parametrize(
        "M",
        [
            1,
        ],
    )
    @parametrize("inplace_add", [True,])
    @parametrize("expand_a_scale", [False,])
    def test_reproduce_issue(
        self, has_bias, dtype, dynamic, reshape_a, M, inplace_add, expand_a_scale
    ):
        r"""
        This testcase check if we can match the int8_dynamic_activation_int8_weight int8 linear pattern from torchao,
        when activation is symmetrically quantized dynamically & weights are symmetrically quantized (statically)
        The pattern is:
            (no bias) _int_mm -> convert_element_type -> ([expand_a] -> mul) -> mul
        or
            (with bias) pattern_no_bias -> add
        Expansion of the scale of activation is optional.
        The pattern depiction doesn't mean that convert_element_type output is fed into expand_a as input,
        but simply that activation scale may be applied after an expand operation on it.
        """
        if dtype == torch.bfloat16 and not torch.ops.mkldnn._is_mkldnn_bf16_supported():
            return
        in_feature = 32
        out_feature = 64
        q_min, q_max = -32, 31
        # we only test for qlinear_binary in this case

        class Mod(torch.nn.Module):
            def __init__(self, dtype: torch.dtype, has_bias: bool):
                super().__init__()
                self.dtype = dtype
                self.has_bias = has_bias
                self.b = torch.randint(
                    q_min, q_max, [in_feature, out_feature], dtype=torch.int8
                )
                self.a_scale = torch.rand([M, 1], dtype=dtype) * 0.01 + 0.01
                self.b_scale = torch.rand([out_feature]) * 0.01 + 0.01
                self.b_scale = self.b_scale.to(dtype)
                self.bias = torch.rand([out_feature], dtype=dtype) if has_bias else None
                self.additive = torch.rand([M, out_feature], dtype=dtype)

            def forward(self, a):
                if reshape_a:
                    a_reshaped = a.reshape(-1, a.size(-1))
                else:
                    a_reshaped = a
                c = torch._int_mm(a_reshaped, self.b)
                c = c.to(self.dtype)
                if expand_a_scale:
                    a_scale = self.a_scale.expand(c.shape)
                else:
                    a_scale = self.a_scale
                c = c * a_scale
                c = c * self.b_scale
                if self.has_bias:
                    c = c + self.bias
                elif inplace_add:
                    # When M is 1, dynamic shapes are enabled with torch.compile, has_bias is False,
                    # expand_a_scale is False and inplace_add is true,
                    # the output's outermost dim's stride can't be determined due to some Dynamo bug.
                    c.add_(self.additive)
                return c

        mod = Mod(dtype, has_bias).eval()
        a = torch.randint(q_min, q_max, [M, in_feature], dtype=torch.int8)

        def matcher_check_fn():
            self.assertEqual(
                counters["inductor"]["qlinear_weight_prepack_matcher_count"], 1
            )

        self._test_common(
            mod,
            (a,),
            matcher_check_fn,
            check_autocast=dtype,
            compile_options={"dynamic": dynamic},
        )
```

### Versions

Main branch

**Comment by leslie-fang-intel** [comment]
Add more detail to reproduce this failure:

- Script to run: https://gist.github.com/leslie-fang-intel/f9686fa15b181d82294861aac1d5abad
- PyTorch: main branch b17f5223a456a9d2ccc2760b759e0f74bbe6de64 after the land of https://github.com/pytorch/pytorch/pull/147033

**Comment by leslie-fang-intel** [comment]
Add more detail to reproduce this failure:

- Script to run: https://gist.github.com/leslie-fang-intel/f9686fa15b181d82294861aac1d5abad
- PyTorch: main branch b17f5223a456a9d2ccc2760b759e0f74bbe6de64 after the land of https://github.com/pytorch/pytorch/pull/147033

Looking into the `fx_graph_transformed.py` with debug flag

```
class <lambda>(torch.nn.Module):
    def forward(self, arg4_1: "Sym(s0)", arg5_1: "i8[1, s0]"):
        # No stacktrace found for following nodes
        arg1_1: "bf16[1, 1]" = self._frozen_param1
        arg3_1: "bf16[1, s1]" = self._frozen_param3
        qlinear_prepack: "i8[32, 64]" = self._frozen_param4
        convert_element_type_default: "f32[64]" = self._frozen_param5
        
         # File: /localdisk/leslie/torch_miniforge/test.py:33 in forward, code: a_reshaped = a.reshape(-1, a.size(-1))
        view: "i8[1, s0]" = torch.ops.aten.reshape.default(arg5_1, [-1, arg4_1]);  arg5_1 = arg4_1 = None
        
         # File: /localdisk/leslie/torch_miniforge/test.py:50 in forward, code: c.add_(self.additive)
        qlinear_binary: "bf16[1, s1]" = torch__inductor_fx_passes_quantization_qlinear_binary(x = view, x_scale = arg1_1, x_zp = None, packed_weight = qlinear_prepack, w_scale = convert_element_type_default, w_zp = None, x_2 = arg3_1, b = None, output_scale = 1.0, output_zero_point = 0, output_dtype = torch.bfloat16, x2_scale = 1.0, x2_zp = 0, binary_op_name = 'sum', alpha = 1.0, unary_op_name = 'none', unary_op_args = [], unary_op_algorithm = '');  view = arg1_1 = qlinear_prepack = convert_element_type_default = arg3_1 = None
        return (qlinear_binary,)
```

The line `arg3_1: "bf16[1, s1]" = self._frozen_param3` looks odd because `self._frozen_param3` is a constant derived from the module attribute `self.additive = torch.rand([M, out_feature], dtype=dtype)`.
Why does Dynamo mark the size as symbolic in this case? cc @anijain2305 Could you kindly help to take a look?
